utils/showcase.bash

```diff
@@ -0,0 +1,68 @@
+#!/bin/bash
+
+# Copyright 2018 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+set -e
+
+# Setup a test workspace. We gitignore this.
+rm -rf showcase-testdir
+mkdir showcase-testdir/
+
+pushd showcase-testdir
+go mod init showcase-test
+
+protoc \
+	--go_gapic_out . \
+	--go_gapic_opt 'cloud.google.com/go/showcase/apiv1alpha2;showcase' \
+	--descriptor_set_in=<(curl -sSL https://github.com/googleapis/gapic-showcase/releases/download/v0.0.7/gapic-showcase-0.0.7.desc) \
+	google/showcase/v1alpha2/echo.proto
+
+# With Go modules, we're transitioning off version package.
+# For now this hack will keep things passing.
+mkdir -p cloud.google.com/go/internal/version
+cat > cloud.google.com/go/internal/version/version.go <<EOF
+package version
+
+const Repo = "UNKNOWN"
+func Go() string {return "UNKNOWN"}
+EOF
+
+# TODO(pongad): Move this file into this repository once we deprecate the old generator.
+curl -sSL https://raw.githubusercontent.com/googleapis/gapic-generator/753ff9d8a04a59962f3b8c2c06cb79be7df344c8/showcase/go/showcase_integration_test.go \
+	> showcase_integration_test.go
+
+# If the package name of the generated package is "a/b/c", we write files under directory "$OUT/a/b/c",
+# where OUT is the directory passed to protoc. Prior to Go 1.11, we set OUT=$GOPATH/src.
+# This works well because that's where we expect to find the package a/b/c.
+#
+# This plays badly with modules however. If our module name is $MODULE, and we set $OUT to the module root,
```

Comment:
Just to clarify: "module root" is `$MODULE/..`? If you could be slightly more explicit, saying for each case where protoc generates, where gapic-generator-go generates, and what the necessary imports look like.....

Why can't we generate the files directly under `$MODULE`?

Reply:
"Module root" is a directory name. $MODULE in this case is the name of the module itself. If you `git clone github.com/googleapis/gapic-generator-go foobar`, your module name is "github.com/googleapis/gapic-generator-go" because that's what `go.mod` declares. Module root is directory "foobar" because that's where the files are. Am I making sense?

We can generate the files directly under $MODULE, but that'd make us behave differently than `protoc-gen-go` does (I think). Since the two programs do similar things, I think we should try to be consistent.

I made the comment a little more concrete. Let me know if this helps.